oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/name/NamespaceRegistryModel.java

```diff
@@ -205,19 +217,40 @@ public NamespaceRegistryModel setMappings(@NotNull Map<String, String> additiona
             String uri = entry.getValue();
             String encodedUri = Namespaces.encodeUri(uri);
 
+            String previousMappedUri = newPrefixToNamespaceMap.get(prefix);
```

Comment:
I'm getting a bit lost here re the newly added code. Maybe you could add a few comments?

Reply:
Yes, see upcoming commit.